OpenRA.Mods.Common/Traits/Render/WithDeathAnimation.cs

```diff
@@ -101,5 +113,21 @@ public void SpawnDeathAnimation(Actor self, string sequence, string palette)
 					w.Add(new Corpse(w, self.CenterPosition, rs.GetImage(self), sequence, palette));
 			});
 		}
+
+		void INotifyCrushed.OnCrush(Actor self, Actor crusher, HashSet<string> crushClasses)
+		{
+			crushed = true;
+
+			if (Info.CrushedSequence == null)
+				return;
```

Comment:
The problem seems to be that 

1. `Mobile.FinishedMoving` doesn't take actors with `SharedCells` into account when calling `OnCrush` (not sure if it should).
2. there is no check equivalent to `Crushable.CrushableBy` happening here.

Also, *if* `Mobile.FinishedMoving` actually calls `OnCrush` that often (ie. every time infantry units share cell), just setting `crushed` here will cause the wrong death animation to play if the unit is later shot, for example.